### ISSUE TITLE
enable rate law expressions to contain parameters

### ISSUE THREAD

**Issue by aesundstrom** [KarrLab/wc_lang#10]
In my `wc_lang` model, `h1_hesc/cell_cycle/tyson_2011/hybrid_model/hybrid_model_rules.py`, I declare a Parameter `Ksa1`:

```
Ksa1 = Parameter(
    id='Ksa1', name='Ksa1', model=H1_hESC_model, submodels=[cell_cycle_submodel],
    value=5 * sf_CycA.value, units='mol/(L*hr)',
    references=[tyson_2011_ref])
```

Later, I declare a Reaction whose `RateLawEquation` expression uses `Ksa1`:

```
Rsa1 = Reaction(
    id='Rsa1', name='Rsa1', submodel=cell_cycle_submodel,
    rate_laws=[RateLaw(direction=forward, 
    equation=RateLawEquation(expression='Ksa1'))])
Rsa1.participants.create(species=CycA, coefficient=1)
```

Later, I run `CheckModel`:

`CheckModel(H1_hESC_model).run()`

Which produces the following error:

```
Traceback (most recent call last):
  File "hybrid_model_rules.py", line 679, in <module>
    CheckModel(H1_hESC_model).run()
  File "/usr/local/lib/python3.6/site-packages/wc_lang/prepare.py", line 678, in run
    raise ValueError('\n'.join(self.errors))
ValueError: Error: NameError in transcoded rate law 'Ksa1 ': 'name 'Ksa1' is not defined'
```

I would like to define a `RateLawEquation` using Parameters.

**Comment by aesundstrom** [comment]
Modeling & Representation meeting notes from 10 Apr 2018:

Arthur raised the point that the Parameter and Species namespaces can collide (e.g., a `RateLawExpression` that contains a Parameter `k` and a Species `k` (for potassium, say) in the `cytosol` Compartment: `equation=RateLawEquation(expression='k*k[cytosol]')`).

Discussion:

Use a global namespace.

Proposed `RateLawExpression` parse spec:

Parameters are stand-alone symbols, whereas Species are immediately followed by `<Compartment id>` in square braces, followed optionally by a comment in curly braces that can denote a hyperlink to `<namespace>:<object name>`:

`<Parameter id>`

`<Species id>[<Compartment id>]{comment that can include hyperlink to <namespace>:<object name>}`

Eliminate `Km` and `Vmax` attributes; instead, use a free-form parsed `RateLawExpression` string that can have an arbitrary number of parameters, including `Km` and `Vmax`.

**Comment by jonrkarr** [comment]
Implemented in 69533de15e36c05c0a6a80089ad2ae5ecaec69f9
